### PR TITLE
Add attachment support

### PR DESCRIPTION
**Pull request by keith**
Closes https://github.com/br1sk/Sonar/issues/1

### CODE REVIEW

File: Sources/Sonar/APIs/AppleRadarRouter.swift [hunk 1]
```diff
@@ -59,7 +60,7 @@ enum AppleRadarRouter {
                     "configurationSplitValue": radar.configuration,
                     "workAroundText": "",
                     "descriptionText": radar.body,
-                    "problemAreaTypeCode": radar.area.map { String($0.appleIdentifier) } ?? "",
+                    "problemAreaTypeCode": (radar.area.map { String($0.appleIdentifier) } ?? "") as Any,
```
Comment (Reflejo) [inline Sources/Sonar/APIs/AppleRadarRouter.swift]: Why do you need this?

File: Sources/Sonar/Models/Radar.swift [hunk 2]
```diff
@@ -36,11 +36,13 @@ public struct Radar {
     public let applicationID: String?
     // Email address or username of the user experiencing the issue (iTunes Connect only)
     public let userID: String?
+    // The URLs of the attachments to send with the radar
```
Comment (Reflejo) [inline Sources/Sonar/Models/Radar.swift]: Not urls

File: Sources/Sonar/APIs/AppleRadarRouter.swift (hunk 1 above)
Comment (keith) [inline Sources/Sonar/APIs/AppleRadarRouter.swift]: Getting a warning without it 🤔

Comment (keith) [inline Sources/Sonar/APIs/AppleRadarRouter.swift]: In 8.2.1:

image

File: Sources/Sonar/Models/Radar.swift (hunk 2 above)
Comment (keith) [inline Sources/Sonar/Models/Radar.swift]: fixed

File: Sources/Sonar/APIs/AppleRadarRouter.swift (hunk 1 above)
Comment (Reflejo) [inline Sources/Sonar/APIs/AppleRadarRouter.swift]: If that's a String? Something else is wrong here :/

Comment (keith) [inline Sources/Sonar/APIs/AppleRadarRouter.swift]: `flatMap` solves it, but it seems like this initializer shouldn't be failable 🤔

Comment (keith) [inline Sources/Sonar/APIs/AppleRadarRouter.swift]: This has to be a bug, if I take this statement out of the dictionary literal it is correctly typed.

Comment (keith) [inline Sources/Sonar/APIs/AppleRadarRouter.swift]: Yep, this was introduced by my sizes ternary below. Here's a pretty small repro case:

```swift
struct Area {
    let appleIdentifier: Int
}

struct Radar {
    let area: Area?

    init() {
        self.area = nil
    }
}

let radar = Radar()
let sizes = [""]
let JSON: [String: Any] = [
    "problemAreaTypeCode": radar.area.map { String($0.appleIdentifier) } ?? "",
    "hiddenFileSizeNew": [].isEmpty ? "" : sizes,
]
```

Comment (keith) [inline Sources/Sonar/APIs/AppleRadarRouter.swift]: Testing this on the beta to see if it's still a problem.

Comment (keith) [inline Sources/Sonar/APIs/AppleRadarRouter.swift]: This is fixed in 8.3 beta 4, I think I'll just undo this work around and let the error be in master.

**Pull request merged**